ironfish/src/rpc/clients/socketClient.ts

```diff
@@ -349,6 +350,9 @@ export abstract class RpcSocketClient extends RpcClient {
 
   protected onError(error: unknown): void {
     this.logger.error(ErrorUtils.renderError(error))
+    for (const request of this.pending.values()) {
+      request.reject(error)
```

Comment:
Curous about Jason's question as well. But.. if we do intend to reject all requests would we also need to close connection to the server?

Reply:
If we get a transmission error, trying to close the connection would result in another error, because transmitting the RST packet would fail too (which is fine, as long as it's properly handled). However, we do need to close the file descriptor--I think it's done automatically, am I wrong?